kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/item/MagicCommandType.java

```diff
@@ -0,0 +1,34 @@
+package com.twosigma.beakerx.kernel.commands.item;
+
+import com.twosigma.beakerx.kernel.commands.MagicCommandFunctionality;
+
+public final class MagicCommandType {
+  private final String command;
+  private final String parameters;
+  private final MagicCommandFunctionality functionality;
+  private final String suitableKernelsNames;
+
+  public MagicCommandType(String command, String parameters,
+      MagicCommandFunctionality functionality, String suitableKernelsNames) {
+    this.command = command;
+    this.parameters = parameters;
+    this.functionality = functionality;
+    this.suitableKernelsNames = suitableKernelsNames;
```

Comment:
this isn't the right way to do this.
instead, have the kernel add the commands.
base kernel adds magics that work in all kernels.